xapian-core/include/xapian/weight.h

```diff
@@ -471,10 +471,12 @@ class XAPIAN_VISIBILITY_DEFAULT TfIdfWeight : public Weight {
      * Implementing support for more normalizations of each type would require
      * extending the backend to track more statistics.
      */
-    explicit TfIdfWeight(double slope, double delta, const std::string &normalizations);
+    explicit TfIdfWeight(const std::string &normalizations);
+
+    explicit TfIdfWeight(const std::string &normalizations, double slope, double delta);
 
     TfIdfWeight()
-	: param_slope(0.2), param_delta(1.0), normalizations("ntn")
+	: normalizations("ntn")
```

Comment:
Slope and delta should be `0.2` and `1.0` respectively by default as proposed in the paper. Updated the default values from the current initialised `0.0` to `0.2` and `1.0`. 

So, in case of `TfIdfWeight("PPP")` default values will be there for slope and delta or if someone wants to use `0.0` then constructing `TfIdfWeight("PPP", 0.0, 0.0)` should be the ideal choice.